drivers/Aqara/aqara-presence-sensor/src/fp2/device_manager.lua

```diff
@@ -0,0 +1,272 @@
+local log = require "log"
+local json = require "st.json"
+local fields = require "fields"
+local capabilities = require "st.capabilities"
+local multipleZonePresence = require "multipleZonePresence"
+
+local PresenceSensor = capabilities.presenceSensor
+local MovementSensor = capabilities.movementSensor
+local ActivitySensor = capabilities.activitySensor
+local DeviceMode = capabilities["stse.deviceMode"]
+
+local MOVEMENT_TIMER = "movement_timer"
+local MOVEMENT_TIME = 5
+local COMP_MODE = "mode"
+
+local device_manager = {}
+device_manager.__index = device_manager
+
+local FP2_MODES = { "zoneDetection", "fallDetection", "sleepMonitoring" }
+
+function device_manager.presence_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+  local evt_action = "not present"
+  if evt_value == 1 then evt_action = "present" end
+  device:emit_event(PresenceSensor.presence(evt_action))
+end
+
+function device_manager.movement_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+
+  local val = evt_value
+  local no_movement = function()
+    if not device:supports_capability(PresenceSensor) then return end
+    device:emit_event(MovementSensor.movement("inactive"))
+  end
+  device:set_field(MOVEMENT_TIMER, device.thread:call_with_delay(MOVEMENT_TIME, no_movement))
+
+  if val == 0 then
+    device:emit_event(MovementSensor.movement("entering"))
+  elseif val == 1 then
+    device:emit_event(MovementSensor.movement("leaving"))
+  elseif val == 2 then
+    device:emit_event(MovementSensor.movement("enteringLeft"))
+  elseif val == 3 then
+    device:emit_event(MovementSensor.movement("leavingRight"))
+  elseif val == 4 then
+    device:emit_event(MovementSensor.movement("enteringRight"))
+  elseif val == 5 then
+    device:emit_event(MovementSensor.movement("leavingLeft"))
+  elseif val == 6 then
+    device:emit_event(MovementSensor.movement("approaching"))
+  elseif val == 7 then
+    device:emit_event(MovementSensor.movement("movingAway"))
+  end
+end
+
+function device_manager.zone_presence_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+
+  local zoneInfo = multipleZonePresence.findZoneById(zone)
+  if not zoneInfo then
+    multipleZonePresence.createZone("zone" .. zone, zone)
```

Comment:
Is it expected that zone names set in the Aqara app are not reflected to SmartThings? I named a zone in the Aqara app but in the SmartThings I still saw "zone1".

Reply:
Yes, zone names in the Aqara Home app are not reflected in the SmartThings app; SmartThings users will need to rename zones manually. (The SmartThings app has an icon to rename the zone.)